src/tool/hpcrun/control-knob.c

```diff
@@ -38,12 +38,16 @@ control_knob_register(char *name, char *value, control_knob_type type)
   if (iter == NULL) {
     iter = (control_knob_t*) malloc(sizeof(control_knob_t));
     iter->name = strdup(name);
-    iter->type = type;
     iter->next = control_knobs;
+    iter->type = type;
     control_knobs = iter;
   }
-  iter->value = strdup(value);
-  
+  if (iter->type == ck_string) {
+    iter->value = strdup(value);
+  } else {
+    iter->value = (char *) malloc(sizeof(char) * (strlen(value) + 1));
+    strcpy(iter->value, value);
+  }
```

Comment:
Completely inconsequential change. Remove.

Reply:
AFAICT, the `else` clause is identical to the implementation of `strdup`. What's the fix?

Reply:
Oh yes, thanks for pointing it out! I was referring to other code in this file but they were merged together.